src/extension.ts

```diff
@@ -58,11 +63,21 @@ import { startTreeView } from "./treeview";
 import { MetalsFeatures } from "./MetalsFeatures";
 import { MetalsTreeViewReveal, MetalsTreeViews } from "./tree-view-protocol";
 import * as scalaDebugger from "./scalaDebugger";
+import {
+  DecorationTypeDidChange,
+  DecorationsRangesDidChange
+} from "./decoration-protocol";
 
 const outputChannel = window.createOutputChannel("Metals");
 const openSettingsAction = "Open settings";
 const openSettingsCommand = "workbench.action.openSettings";
 let treeViews: MetalsTreeViews | undefined;
+let decorationType: TextEditorDecorationType = window.createTextEditorDecorationType(
```

Comment:
Having a single `decorationType` means that the server will need to manage (and mix) multiple decoration sources, is that intentional? I'm fine with it, but I just wanted to check

Reply:
So far, the Metals server doesn't update the `decorationType` but we have an endpoint to change it just in case that will be handy for the future. I wanted to minimize the need for updating the VS Code client once we want to experiment with using larger parts of the decoration API. 

If we end up needing to use multiple decoration types then we could maybe give them somehow names. For now, it's enough to support only a single default decoration type.